sfge/src/main/java/com/salesforce/graph/ops/expander/ApexPathExpanderUtil.java

```diff
@@ -288,29 +289,9 @@ private void expand(
                         LOGGER.warn("expand-StackDepthLimit. ex=" + ex);
                     }
                 } catch (MethodPathForkedException ex) {
-                    List<ApexPathExpander> forkedPathExpanders = new ArrayList<>();
-                    for (ApexPathExpander forkedApexPathExpander : getForkedExpanders(ex)) {
-                        if (forkedApexPathExpander.getTopMostPath().endsInException()) {
-                            // This happens when a path invokes a method that only throws an
-                            // exception
-                            ThrowStatementVertex throwStatementVertex =
-                                    forkedApexPathExpander
-                                            .getTopMostPath()
-                                            .getThrowStatementVertex()
-                                            .get();
-                            logFilteredOutPath(throwStatementVertex);
-                            forkedApexPathExpander.finished();
-                        } else {
-                            apexPathExpanders.push(forkedApexPathExpander);
-                            forkedPathExpanders.add(forkedApexPathExpander);
-                        }
-                    }
-                    apexPathCollapser.pathForked(
-                            ex.getForkEvent(), apexPathExpander, forkedPathExpanders);
-                    apexPathExpander.finished();
-                    if (LOGGER.isInfoEnabled()) {
-                        LOGGER.info("expand-Forked. ex=" + ex);
-                    }
+                    // Add this exception to the PendingForkStack, so it can be lazily processed
+                    // later.
+                    pendingForkStack.addStackFrame(ex);
```

Comment:
I know we spoke about this pattern of using catch for business logic and we all agreed that this is not a good practice. Is there an easy way to fix this as part of this PR?

Reply:
I'm not sure. It's one of the remaining things I'm looking into. My preference would be to merge what we have for now, though, since the current fix gets us to 80% of where we want to be.